src/setup/node.rs

```diff
@@ -1,193 +1,187 @@
+        fs::write(path, content)?;
+        let node = self.start_node(log_to_stdout);
+        wait_for_start(node.addr).await;
+        Ok(node)
+    }
+
+    /// Sets address to bind to.
+    pub fn set_addr(mut self, addr: SocketAddr) -> Self {
+        self.conf.local_addr = addr;
+        self
+    }
+
+    /// Adds arguments to start command.
+    pub fn add_args(mut self, args: Vec<String>) -> Self {
+        args.into_iter()
+            .for_each(|arg| self.meta.start_args.push(arg.into()));
+        self
+    }
 
-        Ok(())
+    /// Sets initial peers for the node.
+    pub fn initial_peers(mut self, addrs: Vec<SocketAddr>) -> Self {
+        self.conf.initial_peers = addrs.into_iter().collect();
+        self
     }
 
-    async fn start_process(&mut self) -> Result<()> {
-        // cleanup any previous runs (node.stop won't always be reached e.g. test panics, or SIGINT)
-        self.cleanup()?;
+    /// Sets validator token to be placed in rippled.cfg.
+    /// This will configure the node to run as a validator.
+    pub fn validator_token(mut self, token: String) -> Self {
+        self.conf.validator_token = Some(token);
+        self
+    }
 
-        // TODO: set initial peers/initial actions.
-        self.generate_config_file()?;
+    /// Sets network's id to form an isolated testnet.
+    pub fn network_id(mut self, network_id: u32) -> Self {
+        self.conf.network_id = Some(network_id);
+        self
+    }
 
-        let (stdout, stderr) = match self.config.log_to_stdout {
+    fn start_node(self, log_to_stdout: bool) -> Node {
+        let (stdout, stderr) = match log_to_stdout {
             true => (Stdio::inherit(), Stdio::inherit()),
             false => (Stdio::null(), Stdio::null()),
         };
-
-        let process = Command::new(&self.meta.start_command)
+        let child = Command::new(&self.meta.start_command)
             .current_dir(&self.meta.path)
             .args(&self.meta.start_args)
             .stdin(Stdio::null())
             .stdout(stdout)
             .stderr(stderr)
             .spawn()
             .expect("node failed to start");
-
-        self.wait_for_start().await;
-        self.process = Some(process);
-
-        Ok(())
-    }
-
-    async fn wait_for_start(&self) {
-        wait_until!(
-            CONNECTION_TIMEOUT,
-            if let Ok(mut stream) = tokio::net::TcpStream::connect(self.addr()).await {
-                stream.shutdown().await.unwrap();
-                true
-            } else {
-                false
-            }
-        );
+        Node {
+            child,
+            addr: self.conf.local_addr,
+        }
     }
+}
 
-    fn generate_config_file(&self) -> Result<()> {
-        let path = self.config.path.join(RIPPLED_CONFIG);
-        let content = RippledConfigFile::generate(&self.config)?;
-        fs::write(path, content)?;
+/// Fields to be written to the node's configuration file.
+#[derive(Debug)]
+pub struct NodeConf {
```

Comment:
Nit: I wouldn't rename this to stay consistent with the rest of the naming used throughout, `NodeConfig` and `config` are only 2 letters more.

Reply:
Done